src/dev/riscv/lupv.cc

```diff
@@ -50,12 +52,14 @@ LupV::LupV(const Params &params) :
 void
 LupV::postConsoleInt()
 {
+    plic->post(lupioTTYIntID);
     plic->post(uartIntID);
 }
```

Comment:
You can't raise interrupts for both interrupts. Otherwise, the software will be completely confused. If the uart calls `postConsoleInt()` then only the related IRQ at the PLIC level should be raised, and same for the lupio-tty.

You can either define another set of postConsoleInt()/clearConsoleInt() (eg with a different naming) that is just for the lupio-tty, or you need to accept an argument that says specifically which one should be raised.

I would suggest making a pair of generic functions, such as `postPicInt/clearPicInt(int intID)`. If each device knows its own intID, then it can call this pair of functions with it. It will be useful for lupio-blk as well.

Reply:
Something I don't understand about the platform code in general: Is there a reason the lupioTTY has to call into the platform to post the interrupt to the plic? Could it just have a pointer to a generic device and call the post() function itself?

Reply:
I guess one drawback of having the device call the `post` function of the interrupt controller directly is that the device needs a pointer to the interrupt controller instance with a type that defines the `post` function. In that case, we're back to the problem of the device being dependent on a certain model of interrupt controller.

A possible solution that may solve this is to have a proper hierarchy of classes. If interrupt controllers derived from a generic interrupt controller class that defines a pair of `post/clear` virtual functions, then devices would only need to have a pointer onto an instance of this generic class and call these functions without any issues.

Or does the BasicPioDevice already define a `post()` function?

Reply:
Ah, that makes sense. No, it doesn't look like there's a virtual `post()` function, which is unfortunate.

I guess you need to add another `LupV::post???Interrupt` function for the lupio TTY. Maybe use `postUartInt()` instead of "console"?

Reply:
Sorry, I left the uart in there because I was switching out the uart and LupIO-TTY when testing and I had intended to remove the uart entirely from the LupV platform. Should I then support both the uart8250 and LupIO-TTY in the LupV platform?

Reply:
Yeah, that's fine for the LupV platform, but now I'm curious as to how this is supposed to work more generally! What if there's two consoles? (I don't think that's impossible in real life...)